elasticdl/pkg/commonnew/tensor.go

```diff
@@ -178,13 +178,35 @@ func (t *Tensor) SerializeToTensor() *tensor_go_proto.TensorProto {
 	}
 }
 
+// IndexedTensor : IndexedSlice in memory representation
+type IndexedTensor struct {
```

Comment:
We could just follow proto message definition, name it to `IndexedSlices`

```go
type IndexedSlices struct {
    ConcatTensors *Tensor
    IDs []int64
}
```

Reply:
```
Tensor -> TensorProto
IndexedSlice -> IndexedSliceProto
```
could this be better?

Reply:
Since it's in memory representation, I think Proto is not needed.

Reply:
And here, we could use `tensorflow.TensorProto` and `proto.IndexedSlices` directly. 

Pros:
- no need to define extra in memory data structures, `Tensor`/`IndexedSlices`
- no need to define extra serialize/deserialize functions, `SerializeTensor`/`DeserializeTensor`/`SerializeIndexedSlices`/`DeserializeIndexedSlices`

Cons:
- tensorflow.TensorShapeProto is not convenient to use directly, we need to add two extra function, `SetDim`/`GetDim`

Reply:
I find that it's not convenient to write codes without defining in memory data structures. So let's move on.

And for the struct name, How about

```go
type struct IndexedSlices {
    ConcatTensors *Tensor
    IDs []int64
} 
```